connectors/src/connectors/intercom/index.ts

```diff
@@ -0,0 +1,127 @@
+import { validateAccessToken } from "@connectors/connectors/intercom/lib/intercom_api";
+import { ConnectorPermissionRetriever } from "@connectors/connectors/interface";
+import { Connector, ModelId, sequelize_conn } from "@connectors/lib/models";
+import { IntercomConnectorState } from "@connectors/lib/models/intercom";
+import { getAccessTokenFromNango } from "@connectors/lib/nango_helpers";
+import { Err, Ok, Result } from "@connectors/lib/result";
+import logger from "@connectors/logger/logger";
+import { DataSourceConfig } from "@connectors/types/data_source_config";
+import { ConnectorsAPIErrorResponse } from "@connectors/types/errors";
+import { NangoConnectionId } from "@connectors/types/nango_connection_id";
+import {
+  ConnectorPermission,
+  ConnectorResource,
+} from "@connectors/types/resources";
+
+const { NANGO_INTERCOM_CONNECTOR_ID } = process.env;
+
+export async function createIntercomConnector(
+  dataSourceConfig: DataSourceConfig,
+  connectionId: NangoConnectionId
+): Promise<Result<string, Error>> {
+  const nangoConnectionId = connectionId;
+
+  if (!NANGO_INTERCOM_CONNECTOR_ID) {
+    throw new Error("NANGO_INTERCOM_CONNECTOR_ID not set");
+  }
+
+  const accessToken = await getAccessTokenFromNango({
+    connectionId: nangoConnectionId,
+    integrationId: NANGO_INTERCOM_CONNECTOR_ID,
+    useCache: false,
+  });
+
+  if (!validateAccessToken(accessToken)) {
+    return new Err(new Error("Intercom access token is invalid"));
+  }
+
+  try {
+    const connector = await sequelize_conn.transaction(async (t) => {
+      const connector = await Connector.create(
+        {
+          type: "intercom",
+          connectionId: nangoConnectionId,
+          workspaceAPIKey: dataSourceConfig.workspaceAPIKey,
+          workspaceId: dataSourceConfig.workspaceId,
+          dataSourceName: dataSourceConfig.dataSourceName,
+          defaultNewResourcePermission: "read_write",
```

Comment:
is this cargo culted from Slack? This is a special case. I thiiink it should be `read` (or make sure it's aligned with Notion/Github/Drive)

Reply:
We use read_write on all connectors as default because once we start having actions that write to these connectors we won't have to migrate them to read_write. Do you feel it should be "read " ?